src/datasets.c

```diff
@@ -532,6 +532,46 @@ Dataset *DatasetGet(const char *name, enum DatasetTypes type,
     return NULL;
 }
 
+int DatasetReload(uint32_t de_vers)
+{
+    Dataset *set = sets;
+    // Reload if its a static set only
+    while (set != NULL) {
+        // Check if the set is static or dynamic
+        if ((set->save == NULL || strcmp(set->save, "") == 0) || (set->from_yaml == false)) {
+            SCLogInfo("It is a static set or not defined in YAML, so we can continue...");
+        } else {
+            continue;
+        }
+        SCLogInfo("Soon to be hidden Dataset name: %s", set->name);
+        SCLogInfo("Soon to be hidden Dataset type: %d", set->type);
+        set->hidden = true;
+        set = set->next;
+    }
+    return 1;
+}
+
+void DatasetRemoveCopies(void)
+{
+    SCLogInfo("destroying old copies of datasets: %p", sets);
+    SCMutexLock(&sets_lock);
+    Dataset *set = sets;
+    while (set) {
+        if (set->hidden == false) {
+            set = set->next;
+            continue;
+        }
+        SCLogInfo("destroying set %s, hidden: %d address: %p", set->name, set->hidden, set);
+        Dataset *next = set->next;
+        THashShutdown(set->hash);
+        SCFree(set);
+        set = next;
+    }
+    SCLogInfo("destroying old copies of datasets done: %p", sets);
+    sets = NULL;
```

Comment:
this doesn't look right: we now no longer have any reference to the sets list

Reply:
the list management here needs a bit more thought. Test with a mixed list of dynamic and static sets and walk it after the cleanup to see it is still consistent.